torch/distributed/_sharded_tensor/__init__.py

```diff
@@ -539,3 +542,11 @@ def wrapper(*args, **kwargs):
 
 # Import all builtin sharded ops
 from .ops import *  # noqa: F403
+
+def reshard_output(
+        module: torch.nn.Module,
+        resharding_spec: ReshardingSpec) -> ModuleResharder:
+    """
+    Replaces a module with a new module that will reshard its output.
```

Comment:
For now, probably make this a private API `_reshard_output` since probably the public API will be at the Sharder level. Also, we need additional docs here in terms of `Args:` and `Returns:`.

nit: "Replaces a module with a new module that will reshard its output according to the given ``resharding_spec``".